### PR TITLE
Create relationship between courses and subject

### PR DESCRIPTION
**Pull request by vjnhat**

### CODE REVIEW

File: app/controllers/supervisor/subject_course_controller.rb [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+  def edit
+  end
+
+  def update
+  end
+
+  def destroy
+  end
+end
```
Comment (brokenheart1712) [inline app/controllers/supervisor/subject_course_controller.rb]: Please follow this order: index, show, new, create, edit, update, destroy

Comment (vjnhat) [inline app/controllers/supervisor/subject_course_controller.rb]: I fixed

File: app/models/course.rb [hunk 2]
```diff
@@ -1,4 +1,6 @@
 class Course < ActiveRecord::Base
+  has_many :subject_courses
+  has_many :subjects, through: :subject_courses
   validates :name, presence:true, length: {maximum: 50}
```
Comment (dieunb) [inline app/models/course.rb]: slice to some block code, to group by type.

File: app/controllers/supervisor/subject_course_controller.rb [hunk 3]
```diff
@@ -0,0 +1,23 @@
+class Supervisor::SubjectCoursesController < ApplicationController
```
Comment (dieunb) [inline app/controllers/supervisor/subject_course_controller.rb]: remove this file. Irrelevant to your this pull

File: config/routes.rb [hunk 4]
```diff
@@ -7,6 +7,7 @@
     resources :users
     resources :subjects
     resources :courses
+    resource  :subject_courses
```
Comment (dieunb) [inline config/routes.rb]: ???

**Pull request merged**